### PR TITLE
TicketTag: Add seperate resourcelist class, docs, permissions for dir…

### LINKED ISSUE #4119: Ticket-tags: Add seperate resourcelist class, docs, permissions for direct endpoints
Parent Issue: #4101.

### PR DESCRIPTION
**Pull request by enigmaeth**
…ect endpoints

Fixes #4119.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by enigmaeth** [comment]
@fossasia/open-event-server-devs Please review!

File: app/api/tickets.py [hunk 1]
```diff
@@ -123,9 +123,7 @@ def before_create_object(self, data, view_kwargs):
             data['event_id'] = event.id
 
     view_kwargs = True
-    decorators = (api.has_permission('is_coorganizer', fetch='event_id',
-                  fetch_as="event_id", model=Ticket, methods="POST",
-                  check=lambda a: a.get('event_id') or a.get('event_identifier')), )
+    decorators = (jwt_required,)
```
Comment (shubham-padia) [inline app/api/tickets.py]: Shouldn't this be is_coorganizer as it was before :smile: ?

File: docs/api/api_blueprint.apib [hunk 2]
```diff
@@ -5518,7 +5518,7 @@ Related to tickets and event.
 |:----------|-------------|------|----------|
 | `name`  | Name of the ticket tag | string | **yes** |
 
-## Ticket Tags Collection [/v1/tickets/{ticket_id}/ticket-tags{?page%5bsize%5d,page%5bnumber%5d,sort,filter}]
+## Ticket Tags Collection [/v1/ticket-tags{?page%5bsize%5d,page%5bnumber%5d,sort,filter}]
 + Parameters
     + ticket_id: 1 (string) - id of the ticket.
```
Comment (shubham-padia) [inline docs/api/api_blueprint.apib]: `ticket_id` can be removed :)

File: app/api/tickets.py (hunk 1 above)
Comment (enigmaeth) [inline app/api/tickets.py]: Had changed to test something, Forgot reverting this before pushing :sweat:

Comment (enigmaeth) [inline app/api/tickets.py]: Resolved.

File: docs/api/api_blueprint.apib (hunk 2 above)
Comment (enigmaeth) [inline docs/api/api_blueprint.apib]: Resolved.

**Pull request merged**